### DISCUSSION TITLE
How to disable rendering within the template block?

### DISCUSSION THREAD

**Discussion by YU000jp** [stdword/logseq13-full-house-plugin#18] [Q&A] [answered]
Is it possible to add an option to prevent rendering when placing rendering within template blocks, similar to SmartBlocks plugin? I would like to implement a branching process for weekdays and holidays when the journal template is called.
https://github.com/sawhney17/logseq-smartblocks#types-of-expansion

**Comment by stdword** [comment, marked as answer]
Hello!

## There are several ways now:
<img width="400px" src="https://github.com/stdword/logseq13-full-house-plugin/assets/1984175/9ecb437f-bbab-47b8-b9a7-7ad06a38fa22"/>


### way 3
- Including another template with [`include`](https://stdword.github.io/logseq13-full-house-plugin/#/reference__tags?id=nesting-include) template tag.

### way 4
- Explicit argument to delay rendering: [`:delay-until-rendered`](https://stdword.github.io/logseq13-full-house-plugin/#/reference__configuring?id=delay-until-rendered):

```clojure
{{renderer :template, lazy, :delay-until-rendered}}
```

**Comment by YU000jp** [comment]
We have found a processing method.
https://github.com/sawhney17/logseq-smartblocks/blob/main/src/index.tsx#L36


https://github.com/YU000jp/logseq-plugin-weekdays-and-weekends
I have today created a plugin that switches journal templates between weekdays and holidays (although I haven't published it yet). However, this plugin only has basic functions. Those who are looking for advanced features will need to use it in conjunction with "Full House" plugin.

**Reply by stdword** [reply]
Hello, @YU000jp! I guess I've faced restriction, which you talked about: the case when we need to use standard logseq renderer.
So I've added special param `:delay-until-rendered`:
```edn
{{renderer :template, journal-embed, :delay-until-rendered}}
```
Need to specify it explicitly to left the user to decide: render template or not.
Link to [documentation](https://stdword.github.io/logseq13-full-house-plugin/#/reference__configuring?id=delay-until-rendered).
